SampleApp/SampleApp/Delegates/AppDelegate.swift

```diff
@@ -39,6 +39,28 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         print("Failed to register for remote notification: \(error.localizedDescription)")
     }
 
+    func userNotificationCenter(_ center: UNUserNotificationCenter,
```

Comment:
Can `didReceive` and `willPresent` both fire for one notification?

Reply:
`didReceive` will get triggered when user opened notifications no matter which app state is (inactive/ foreground/ background ).
And `willPresent` get triggered only when app is in the foreground and device received notifications.
But `willPresent` is duplicated with the NotificationServiceExtension's `didReceive` function, so I will remove this `willPresent` as it's redundant. Sorry, I overlook this one. Thank you!